docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,258 @@
+
+3. **Enhance `etcd-backup-restore`**  
+   - Introduce new CLI commands:
+     - **`reupload-snapshot`** for re-uploading snapshots.  
+     - **`garbage-collect`** for removing older backups whose immutability period has expired.
+
+By capturing a final full snapshot before hibernation, periodically re-uploading it to preserve immutability, and removing stale backups, etcd backups remain safeguarded against accidental or malicious alterations until the cluster is resumed.
+
+##### Etcd CR API Changes
+
+A new field is introduced in `Etcd.spec.backup.store` to indicate the immutability strategy:
+
+```go
+// StoreSpec defines parameters for storing etcd backups.
+type StoreSpec struct {
+    // ...
+    // Immutability configuration for the backup store.
+    Immutability *ImmutabilitySpec `json:"immutability,omitempty"`
+}
+
+// ImmutabilitySpec defines immutability settings.
+type ImmutabilitySpec struct {
+    // RetentionType indicates the type of immutability approach. For example, "Bucket".
+    RetentionType string `json:"retentionType,omitempty"`
+}
+```
+
+If `immutability` is not specified, `etcd-druid` will assume that the bucket is mutable, and no immutability-related logic applies.
+
+##### `etcd-backup-restore` Enhancements
+
+The authors propose adding two new commands to the `etcd-backup-restore` CLI (`etcdbrctl`) to maintain immutability during hibernation and to clean up older snapshots:
+
+1. **`reupload-snapshot`**
+   - Downloads the latest full snapshot from the object store.
+   - Renames the snapshot (for instance, updates its Unix timestamp) to avoid overwriting an existing immutable snapshot.
+   - Uploads the renamed snapshot back to object storage, thereby restarting its immutability timer.
+
+2. **`garbage-collect`**
+   - Scans the object store for older snapshots and deletes them if their immutability period has expired and they are no longer needed, following the standard [garbage collection policy](https://github.com/gardener/etcd-backup-restore/blob/master/docs/usage/garbage_collection.md).
+
+##### etcd Controller Enhancements
+
+When a hibernation flow is initiated (by external tooling or higher-level operators), the [etcd controller](https://github.com/gardener/etcd-druid/blob/master/docs/development/controllers.md#etcd-controller) can:
+
+1. Remove etcd’s client ports (2379/2380) from the etcd Service to block application traffic.
+2. Trigger an [on-demand full snapshot](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/05-etcd-operator-tasks.md#trigger-on-demand-fulldelta-snapshot) via an `EtcdOperatorTask`.
+3. Scale down the `StatefulSet` replicas to zero, provided the previous snapshot step is successful.
+4. Create the `ExtendFullSnapshotImmutabilityTask` if `etcd.spec.backup.store.immutability.retentionType` is `"Bucket"` and based on `etcd.spec.backup.fullSnapshotSchedule`.
+
+##### Operator Task Enhancements
+
+The `ExtendFullSnapshotImmutabilityTask` will create a cron job that:
```

Comment:
what you have missed is that this task should only be active as long as the cluster is hibernated. Once the etcd cluster comes out of hibernation then this task should no longer exist. So have a section on lifecycle of this task.

Reply:
Yes, Added the lifecycle as part of https://github.com/gardener/etcd-druid/pull/884/commits/6e4511a5afcb65ebe8ef391826b7ec1769411853#diff-981f3bcbeed277184597de9af59f3e40fa0c76a43f84c739a0d17fd3948d2f8aR206-R219